bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/connection/Clip2Bridge.java

```diff
@@ -0,0 +1,1049 @@
+     * @throws ApiException if anything fails.
+     */
+    public Resources getResources(ResourceReference reference) throws ApiException {
+        sleepDuringRestart();
+        if (onlineState == State.CLOSED) {
+            throw new ApiException("getResources() offline");
+        }
+        try {
+            return getResourcesImpl(reference);
+        } catch (HttpUnauthorizedException e) {
+            throw new ApiException("getResources() unauthorized error", e);
+        }
+    }
+
+    /**
+     * Internal method to send an HTTP 2 GET request to the Hue Bridge and process its response.
+     *
+     * @param reference the Reference class to get.
+     * @return a Resource object containing either a list of Resources or a list of Errors.
+     * @throws ApiException if the communication failed, or an unexpected result occurred.
+     * @throws HttpUnauthorizedException if the request was refused as not authorised or forbidden.
+     */
+    private Resources getResourcesImpl(ResourceReference reference) throws ApiException, HttpUnauthorizedException {
+        Session session = http2Session;
+        if (Objects.isNull(session) || session.isClosed()) {
+            throw new ApiException("HTTP 2 session is null or closed");
+        }
+        throttle();
+        String url = getUrl(reference);
+        HeadersFrame headers = prepareHeaders(url, MediaType.APPLICATION_JSON);
+        logger.trace("GET {} HTTP/2", url);
+        try {
+            Completable<@Nullable Stream> streamPromise = new Completable<>();
+            ContentStreamListenerAdapter contentStreamListener = new ContentStreamListenerAdapter();
+            session.newStream(headers, streamPromise, contentStreamListener);
+            // wait for stream to be opened
+            Objects.requireNonNull(streamPromise.get(TIMEOUT_SECONDS, TimeUnit.SECONDS));
+            // wait for HTTP response contents
+            String contentJson = contentStreamListener.awaitResult();
+            String contentType = contentStreamListener.getContentType();
+            logger.trace("HTTP/2 200 OK (Content-Type: {}) << {}", contentType, contentJson);
+            if (!MediaType.APPLICATION_JSON.equals(contentType)) {
+                throw new ApiException("Unexpected Content-Type: " + contentType);
+            }
+            try {
+                Resources resources = Objects.requireNonNull(jsonParser.fromJson(contentJson, Resources.class));
+                if (logger.isDebugEnabled()) {
+                    resources.getErrors().forEach(error -> logger.debug("Resources error:{}", error));
+                }
+                return resources;
+            } catch (JsonParseException e) {
+                throw new ApiException("Parsing error", e);
+            }
+        } catch (ExecutionException e) {
+            Throwable cause = e.getCause();
+            if (cause instanceof HttpUnauthorizedException) {
+                throw (HttpUnauthorizedException) cause;
```

Comment:
```suggestion
            if (cause instanceof HttpUnauthorizedException unauthorizedCause) {
                throw unauthorizedCause;
```

Reply:
No. Not yet. See my comment about v3.4.x elswhere.